src/_pytest/_code/code.py

```diff
@@ -208,6 +208,45 @@ def with_repr_style(
     def lineno(self) -> int:
         return self._rawentry.tb_lineno - 1
 
+    def get_python_framesummary(self) -> traceback.FrameSummary:
+        # Python's built-in traceback module implements all the nitty gritty
+        # details to get column numbers of out frames.
+        stack_summary = traceback.extract_tb(self._rawentry, limit=1)
+        return stack_summary[0]
+
+    # Column and end line numbers introduced in python 3.11
+    if sys.version_info < (3, 11):
+
+        @property
+        def end_lineno_relative(self) -> int | None:
+            return None
+
+        @property
+        def colno(self) -> int | None:
+            return None
+
+        @property
+        def end_colno(self) -> int | None:
+            return None
+    else:
+
+        @property
+        def end_lineno_relative(self) -> int | None:
+            frame_summary = self.get_python_framesummary()
+            if frame_summary.end_lineno is None:
+                return None
```

Comment:
Hmm, looks like we need tests covering this branch - though I'm not sure there's a nice way to get that, so maybe a `# pragma: no cover` instead?  Hmm.

Reply:
Yeah, it would be tough to disable the column information to test this branch. Added a no cover.